src/features/headless-checkout/headless-checkout.ts

```diff
@@ -500,6 +501,15 @@ export class HeadlessCheckout {
       };
     }
 
+    if (
+      isApplePaySettingsGuard(configuration) &&
+      !configuration.paymentMethodSettings
```

Comment:
i found this construction is not universal, partner can send paymentMethodSettings from creditCard to another pid (apple pay by example) and our defaults will not work. I think we should check direct flags in paymentMethodSettings